scripts/eqtl_hail_batch/launch_generate_eqtl_spearman.py

```diff
@@ -0,0 +1,149 @@
+"""
+Launch analysis runner for all cell types and chromosomes
+
+For example:
+
+    python3 scripts/hail_batch/eqtl_hail_batch/launch_generate_eqtl_spearman.py \
+        --input-path "gs://cpg-tob-wgs-test/scrna_seq/grch38_association_files" \
+        --dry-run \
+        --output-dir gs://cpg-tob-wgs-test/eqtl_output \
+        --chromosomes '1 2'
+"""
+
+import logging
+import os
+import click
+
+from analysis_runner.cli_analysisrunner import run_analysis_runner
+from google.cloud import storage
+
+
+# def run_analysis_runner(*args, **kwargs):
+#     """Run analysis runner for all cell types and chromosomes"""
+#     print(args, kwargs)
+
+
+@click.command()
+@click.option(
+    '--cell-types',
+    default=None,
+    help='List of cell types to test',
+)
+@click.option(
+    '--chromosomes',
+    required=True,
+    help='List of chromosome numbers to run eQTL analysis on. Space separated, as one argument',  # noqa: E501; pylint: disable=line-too-long
+)
+@click.option(
+    '--input-path',
+    required=True,
+    help=(
+        'A path prefix of where input files are located, eg: gs://MyBucket/folder/. '
+    ),
+)
+@click.option(
+    '--output-dir',
+    required=True,
+    help='A path of where to output files, eg: gs://MyBucket/output-folder/',
+)
+@click.option('--dry-run', is_flag=True, help='Just check if files exist')
+def submit_eqtl_jobs(
+    chromosomes, input_path, output_dir, dry_run=False, cell_types=None
+):
+    """Run association script for all chromosomes and cell types"""
+
+    assert output_dir.startswith('gs://') and input_path.startswith('gs://')
+    chromosomes = chromosomes.split(' ')
+    assert isinstance(chromosomes, list)
+
+    cs_client = storage.Client()
+
+    bucket_name = input_path.split('gs://')[1].split('/')[0]
+    bucket = cs_client.get_bucket(bucket_name)
+    bucket_path = input_path.split(f'gs://{bucket_name}/')[-1]
+
+    def file_exists(files):
+        file_status = storage.Blob(
+            bucket=bucket, name=files.split(bucket_name)[-1].strip('/')
+        ).exists(cs_client)
+        return file_status
+
+    if cell_types is None:
+        # not provided (ie: use all cell types)
+        # we can infer the cell types from the 'expression_files'
+        # subdirectory of the input_path
+        # eg: {cell_type}_expression.tsv
+
+        path_to_expression_files = os.path.join(bucket_path, 'expression_files')
+        logging.info(f'Going to fetch cell types from {path_to_expression_files}')
+        blobs = bucket.list_blobs(prefix=path_to_expression_files + '/', delimiter='/')
+
+        cell_types = [
+            os.path.basename(b.name)[:-15]
```

Comment:
[nitpicking, feel free to ignore]

15 is the length of `_expression.tsv'`? Could be error-prone. I'd assign `_expression.tsv'` to a variable and use `len()`, e.g.:

```
ending = '_expression.tsv''
os.path.basename(b.name)[:-len(ending)]
```

Reply:
That's a really good point. Hard-coding is not ideal, so this is a much better way. I've updated this now. Thanks!